### PR TITLE
Day 4 Assignment Submission

### PR DESCRIPTION
**Pull request by meghanamurthysn**
Sorry for the delay in submitting.

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -10,19 +10,32 @@
...
+def rand_int():
+	rand_num = random.randint(1, 25)
+	n = 5
+	for count in range(n):
+		try:
+			user_inp = int(input("Guess the number: "))
+			if user_inp == rand_num:
+				print("Yay! You guessed the number correctly.")
+				break
+			elif user_inp > rand_num and user_inp <= 25:
```
Comment (danielsgriffin) [inline D04_ex00.py]: Interesting conciseness but how readable is it?

File: HW04_ch07_ex02.py [hunk 2]
```diff
@@ -21,15 +21,24 @@
...
-
+def eval_loop():
+	user_inp = ''
+	while(True):
+		user_inp = input("Enter the expression to be evaluated: ")
+		try:
+			if user_inp != 'done':
+				eval_inp = eval(user_inp)
+				print(eval_inp)
+			else:
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: What if the first input is 'done'?

File: HW04_ch08_ex00.py [hunk 3]
```diff
@@ -17,18 +17,23 @@
...
 # Imports
-
-
 # Body
-
+def count(word,ltr):
+	count = 0
+	for letter in word:
+		if letter == ltr:
+			count = count + 1
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: count += 1

File: HW04_ch08_ex04.py [hunk 4]
```diff
@@ -25,6 +26,7 @@ def any_lowercase1(s):
 
 def any_lowercase2(s):
     """Explain what is wrong, if anything, here.
+    The function correctly returns 'True' (string) after it encounters the first character that is in smaller case and does not evaluate the remaining characters in the string
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: Hmmm. 
Try: `print(any_lowercase2("NOLOWERCASEHERE"))

File: HW04_ch08_ex05.py [hunk 5]
```diff
@@ -2,3 +2,28 @@
...
+def rot_letter(letter, n):
+	""" Rotate a letter by n places"""
+	if letter.islower():
+		return chr((ord(letter) - ord('a')+n) % 26 + ord('a'))
+	elif letter.isupper():
+		return chr((ord(letter) - ord('A')+n) % 26 + ord('A'))
+	else:
+		return letter
+
+def rot_word(word, n):
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: rotate_word

**Comment by danielsgriffin** [comment]
Good job. Make sure you test all your functions for edge cases.

File: HW04_string2.py [hunk 6]
```diff
@@ -41,10 +48,13 @@ def not_bad(s):
 # e.g. 'abcde', the front half is 'abc', the back half 'de'.
 # Given 2 strings, a and b, return a string of the form
 #  a-front + b-front + a-back + b-back
+import math
```
Comment (malavikasrinivasan) [inline HW04_string2.py]: Imports are better placed in the beginning of the script, says PEP-8 :)

File: HW04_ch08_ex00.py [hunk 7]
```diff
@@ -17,18 +17,23 @@
...
+	print("The letter", ltr, " appears", count, "times in", word)
```
Comment (malavikasrinivasan) [inline HW04_ch08_ex00.py]: FYI - comma separated strings in the print statement are already separated by a space

File: HW04_ch08_ex05.py [hunk 8]
```diff
@@ -2,3 +2,28 @@
...
+################################################################################
+def rot_letter(letter, n):
+	""" Rotate a letter by n places"""
+	if letter.islower():
+		return chr((ord(letter) - ord('a')+n) % 26 + ord('a'))
+	elif letter.isupper():
+		return chr((ord(letter) - ord('A')+n) % 26 + ord('A'))
+	else:
+		return letter
+
```
Comment (malavikasrinivasan) [inline HW04_ch08_ex05.py]: So clever!

Comment (malavikasrinivasan) [inline HW04_ch08_ex05.py]: But tooooo awesome for normal people, so please explain with comments!
